### ISSUE TITLE
Admin does not use certificate generated by acme.

### ISSUE THREAD

**Issue by peterbaumert** [ioBroker/ioBroker.admin#1977]
**Describe the bug**  
I created a certificate with acme adapter and set it in admin adapter. yet admin adapter still uses iobroker self signed certificate.

**To Reproduce**  
1. Go to 'acme' adapter and generate certificate with collection name 'iobroker'
2. Go to 'admin' adapter and choose 'ioboker'
3. Restart adapter (does it automatically on save)

**Expected behavior**  
Adapter uses Lets Encrypt certificate

**Screenshots & Logfiles**  
```
2023-07-10 15:25:48.932  - info: admin.0 (1502638) starting. Version 6.6.0 in /opt/iobroker/node_modules/iobroker.admin, node: v16.20.0, js-controller: 5.0.7
2023-07-10 15:25:48.960  - error: admin.0 (1502638) Cannot configure secure web server, because no certificates found: , ,
2023-07-10 15:25:48.961  - info: admin.0 (1502638) requesting all states
2023-07-10 15:25:48.962  - info: admin.0 (1502638) requesting all objects
2023-07-10 15:25:49.127  - info: admin.0 (1502638) Request actual repository...
2023-07-10 15:25:49.357  - info: host.iobroker Updating repository "beta" under "http://download.iobroker.net/sources-dist-latest.json"
2023-07-10 15:25:49.486  - info: admin.0 (1502638) received all objects
2023-07-10 15:25:49.583  - info: admin.0 (1502638) Secure socket.io server listening on port 443
2023-07-10 15:25:49.584  - info: admin.0 (1502638) https server listening on port 443
2023-07-10 15:25:49.585  - info: admin.0 (1502638) Use link "https://127.0.0.1:443" to configure.
```

**Versions:**  
 - Adapter version: 6.6.0
 - JS-Controller version: 5.0.7
 - Node version: v16.20.0
 - Operating system: Debian

**Comment by foxriver76** [comment]
is it really not using it or is it just the logging?

**Comment by peterbaumert** [comment]
> is it really not using it or is it just the logging?

It really does not use it at all.

**Comment by foxriver76** [comment]
Yes, admin won’t allow the selection of certificates anymore. Since change to Webserver module the config had no affect anymore. Webserver should automatically choose a suitable certificate. For details please ask @raintonr

**Comment by peterbaumert** [comment]
So this isnt an issue of admin adapter but webserver module?

Little confusing for enduser then where to open the issue 😄

**Comment by foxriver76** [comment]
Can you explain what the problem is with the certificate, which has been chosen?

webserver Repo: https://github.com/ioBroker/webserver

**Comment by peterbaumert** [comment]
I used acme adapter to create a certificate:

![image](https://github.com/ioBroker/ioBroker.admin/assets/1212934/eac1428c-4c7c-4061-a557-b45921fd8b77)

then I was expecting admin to use it but it does use default one:

![image](https://github.com/ioBroker/ioBroker.admin/assets/1212934/6444e26e-bcdc-4069-a61e-9b86de3edf1e)

**Comment by raintonr** [comment]
Does the domain you told ACME to generate a certificate for match the domain you actually access the site with. IP addresses and aliases don't work. The domain must match.

If the webserver module can't find an ACME generated certificate that matches the FQDN (or wildcard match) the request comes in on it reverts to using the self signed certificate you are seeing.

**Comment by peterbaumert** [comment]
> Does the domain you told ACME to generate a certificate for match the domain you actually access the site with. IP addresses and aliases don't work. The domain must match.
> 
> If the webserver module can't find an ACME generated certificate that matches the FQDN (or wildcard match) the request comes in on it reverts to using the self signed certificate you are seeing.

yes its both exactly the same. `iobroker.mydomain.de`

**Comment by raintonr** [comment]
Is this error still coming out:

```
Cannot configure secure web server, because no certificates found: , ,
```

That can't be helping!

**Comment by foxriver76** [comment]
> Is this error still coming out:
> 
> ```
> Cannot configure secure web server, because no certificates found: , ,
> ```
> 
> That can't be helping!

This should no longer occur in 6.7.1 and above

**Comment by peterbaumert** [comment]
> Is this error still coming out:
> 
> 
> 
> ```
> 
> Cannot configure secure web server, because no certificates found: , ,
> 
> ```
> 
> 
> 
> That can't be helping!

Its not anymore since the update. In INFO logging nothing shows besides that it listens on port 443. in debug it outputs twice the acme certificate collection and then the listening stuff. Nothing else.

**Comment by raintonr** [comment]
> ... in debug it outputs twice the acme certificate collection and then the listening stuff...

Please show us that.

**Comment by peterbaumert** [comment]
I deleted the default certificates, so he cant use them anymore now i get
```
2023-08-10 21:12:29.720  - error: admin.0 (1777745) Cannot configure secure web server, because no certificates found: defaultPublic, defaultPrivate,
2023-08-10 21:12:29.720  - info: admin.0 (1777745) Default certificates seem to be configured but missing. You can execute "iobroker cert create" in your shell to create these.
2023-08-10 21:12:29.721  - error: admin.0 (1777745) Cannot create web-server: Error: Not exists
```

although i still have my collection "iobroker" with status ok

**Comment by raintonr** [comment]
Gotta say, sounds like you have an old/broken version installed. I note your original log says v6.6.0 but 6.7.2 is the latest. Have you tried that?

**Comment by peterbaumert** [comment]
Originally it was that yes. But since 6.7.2 is out i installed it directly and is still the same. Even completely removed all admin and acme instances and added them new again. Same story.

**Comment by peterbaumert** [comment]
> I deleted the default certificates, so he cant use them anymore now i get
> 
> ```
> 2023-08-10 21:12:29.720  - error: admin.0 (1777745) Cannot configure secure web server, because no certificates found: defaultPublic, defaultPrivate,
> 2023-08-10 21:12:29.720  - info: admin.0 (1777745) Default certificates seem to be configured but missing. You can execute "iobroker cert create" in your shell to create these.
> 2023-08-10 21:12:29.721  - error: admin.0 (1777745) Cannot create web-server: Error: Not exists
> ```
> 
> although i still have my collection "iobroker" with status ok

this error seems to come from https://github.com/ioBroker/ioBroker.js-controller/blob/0eb416fc6fe85ca6c85b0547c61ddf9575bf26e3/packages/adapter/src/lib/adapter/adapter.ts#L2344

**Comment by raintonr** [comment]
Guys, there is no need for a user to ever select a certificate. The webserver module chooses one accordingly, either for the requested domain when available or by fallback to the self signed set.

Why is @peterbaumert able to choose something? If there's still that option in the admin interface it should be removed.

**Comment by foxriver76** [comment]
There are cases webserver applies these https://github.com/ioBroker/webserver/blob/abaa9f0421dbd44c11e2ea5405652067ad69dfef/src/lib/webServer.ts#L238

would be much more interesting to get the actual problem fixed, that no correct certificate is chosen automatically in this case here from my point of view.

**Comment by Diginix** [comment]
There is another cert issue related to admin basic settings with a dependency to e.g. simple-api adapter.
https://github.com/ioBroker/ioBroker.simple-api/issues/172
The whole "reworking" of ssl cert management affects obviously a lot more than at first glance.

**Comment by raintonr** [comment]
> The whole "reworking" of ssl cert management affects obviously a lot more than at first glance.

Yes and no. If adapters don't change to use the webserver package then nothing will be affected.
